lib/github.rb

```diff
@@ -99,10 +99,34 @@ def group_project(cohort_students, data, repo)
     contributors = make_request(url)
     github_usernames = cohort_students.map{ |stud| stud.github_name.downcase }
 
-    contributor_usernames = contributors.map { |c| c['login'].downcase }
+    contributor_usernames = contributors.map { |c| c["login"].downcase }
     contributor_usernames << data["user"]["login"].downcase
     contributor_usernames.uniq!
 
+    _, repo_name = repo.repo_url.split("/")
+
+    # Only trigger if we don't have a full pair.
+    if repo_name && contributor_usernames.length < 2
+      pr_title = data["title"].strip
+      name_title = pr_title
+                     .sub(/[[:punct:]\s]*(?:#{cohort_name})[[:punct:]\s]*/i, '')
+                     .sub(/[[:punct:]\s]*(?:#{repo_name})[[:punct:]\s]*/i, '')
+      names = name_title.split(/\s?(?:(?:[&+,;]+)|(?: and )|(?: - ))\s?/)
+
+      names.each do |name|
+        # Get all the students that have this name as part of their name.
+        matches = cohort_students.select { |s| s.name =~ /#{name}/i }
+
+        # Do a word-match if we got multiple matches.
+        if matches.length > 1
+          matches = cohort_students.select { |s| s.name.downcase.split(/\s/).include?(name.downcase) }
+        end
+
+        # Add the user if the name was unambiguous.
+        contributor_usernames << matches.first.github_name if matches.length == 1
+      end
```

Comment:
If it wasn't, could you drop a log line? Might help with future debugging.

Reply:
I just added a bunch of debug logging up above.